src/managers/ThreadManager.js

```diff
@@ -0,0 +1,233 @@
+   * * `10080` (7 days)
+   * @typedef {number} ThreadAutoArchiveDuration
+   */
+
+  /**
+   * Options for creating a thread
+   * @typedef {Object} ThreadCreateOptions
+   * @property {string} name The name of the new Thread
+   * @property {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
+   * @property {MessageResolvable} [startMessage] The message to start a public or news thread from,
+   * creates a private thread if not provided
+   * @param {string} [reason] Reason for creating the thread
+   */
+
+  /**
+   * Creates a new thread in the channel.
+   * @param {ThreadCreateOptions} [options] Options
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Create a new public thread
+   * channel.threads
+   *   .create({
+   *     name: 'food-talk'
+   *     autoArchiveDuration: 60,
+   *     startMessage: channel.lastMessageId,
+   *     reason: 'Needed a separate thread for food',
+   *   })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * // Create a new private thread
+   * channel.threads
+   *   .create({ name: 'mod-talk', autoArchiveDuration: 60, reason: 'Needed a separate thread for moderation' })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async create({ name, autoArchiveDuration, startMessage, reason } = {}) {
+    let path = this.client.api.channels(this.channel.id);
+    if (startMessage) {
+      startMessage = this.channel.messages.resolveID(startMessage);
+      path = path.messages(startMessage);
+    }
+
+    const data = await path.threads.post({
+      data: {
+        name,
+        auto_archive_duration: autoArchiveDuration,
+      },
+      reason,
+    });
+
+    return this.client.actions.ThreadCreate.handle(data).thread;
+  }
+
+  /**
+   * The options for fetching multiple threads, the properties are mutually exclusive
+   * @typedef {Object} FetchThreadsOptions
+   * @property {FetchArchivedThreadOptions} [archived] The options used to fetch archived threads
+   * @property {boolean} [active] When true, fetches active threads. If archived is set, this is ignored!
+   */
+
+  /**
+   * Obtains a thread from Discord, or the channel cache if it's already available.
+   * @param {ThreadChannelResolvable|FetchThreadsOptions} [options] If a ThreadChannelResolvable, the thread to fetch.
+   * If undefined, fetches all active threads.
+   * If an Object, fetches the specified threads.
+   * @param {boolean} [cache=true] Whether to cache the new thread object if it isn't already
+   * @param {boolean} [force=false] Whether to skip the cache check and request the API,
+   * only applies when fetching a single thread
+   * @returns {Promise<?(ThreadChannel|FetchedThreads)>}
+   * @example
+   * // Fetch a thread by its id
+   * channel.threads.fetch('831955138126104859')
+   *   .then(channel => console.log(channel.name))
+   *   .catch(console.error);
+   */
+  fetch(options, cache = true, force = false) {
```

Comment:
With #5824 being merged, this should also use `BaseFetchOptions`.
It looks like this PR is already rebased on-top of a commit including that PR's changes. :eyes: